packages/react-components/src/checkbox/src/Checkbox.tsx

```diff
@@ -1,79 +1,90 @@
 import "./Checkbox.css";
 
 import { Box } from "../../box";
+import { ElementType, ForwardedRef, ReactNode, SyntheticEvent, useMemo } from "react";
+import { InteractionStatesProps, forwardRef, mergeProps, omitProps, resolveChildren, useCheckableProps, useEventCallback, useSlots } from "../../shared";
 import { Text } from "../../text";
 import { VisuallyHidden } from "../../visually-hidden";
-import { any, bool, elementType, func, number, oneOf, oneOfType, string } from "prop-types";
 import { embeddedIconSize } from "../../icons";
-import { forwardRef, useMemo } from "react";
 import { isNil } from "lodash";
-import { mergeProps, omitProps, resolveChildren, useCheckableProps, useEventCallback, useSlots } from "../../shared";
 import { useCheckbox } from "./useCheckbox";
 import { useFieldInputProps } from "../../field";
 import { useToolbarProps } from "../../toolbar";
 
-const propTypes = {
+export interface InnerCheckboxProps extends InteractionStatesProps {
     /**
-     * A controlled checked state value.
+     * @ignore
      */
-    checked: bool,
+    name?: string;
+    /**
+     * @ignore
+     */
+    tabIndex?: number;
+    /**
+    * A controlled checked state value.
+    */
+    checked?: boolean;
     /**
      * The initial value of `checked` when uncontrolled.
      */
-    defaultChecked: bool,
+    defaultChecked?: boolean;
     /**
      * A controlled indeterminate state value.
      */
-    indeterminate: bool,
+    indeterminate?: boolean;
     /**
      * The initial value of `indeterminate`.
      */
-    defaultIndeterminate: bool,
+    defaultIndeterminate?: boolean;
     /**
      * The value to associate with when in a group.
      */
-    value: oneOfType([string, number]),
+    value?: string | number;
```

Comment:
I forgot why a checkbox accept a "number". 

[MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox) specify that a checkbox `value` prop is a DOMString so it doesn't make sense.

I think we should only accept a "string" as value.

Is this something you would to take care of?

Reply:
i don't mind doing the change in this PR. However, according to the doc you linked, any value other than DOMString will be stringified, so it's not exactly true that it only accept a string. According to Typescript's typing of an input, the allowed values are : `value?: string | ReadonlyArray<string> | number;`